lib/https.js

```diff
@@ -34,6 +34,12 @@ const { urlToOptions, searchParamsSymbol } = require('internal/url');
 function Server(opts, requestListener) {
   if (!(this instanceof Server)) return new Server(opts, requestListener);
 
+  if (typeof opts === 'function') {
+    requestListener = opts;
+    opts = undefined;
+  }
+  opts = opts ? util._extend({}, opts) : {};
```

Comment:
IMHO this is a "cold spot" so I opt for standard 🤷‍♂️ 
Any way this could be simply
```
opts = util._extend({}, opts);
```
@XadillaX I think you're free to decide